### PR TITLE
Send new users to Blink.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This pull request adds Blink integration, starting by sending new user profiles to Blink whenever we create an account, so that we can keep accurate records in [Customer.io](https://customer.io).

#### How should this be reviewed?
Check things out commit-by-commit and see the included test case.

#### Checklist
- [ ] Documentation added for changed endpoints.
- [x] Tests added for new features/bug fixes.
- [ ] Post a message in #api if this includes something that causes a rebuild!

### CODE REVIEW

**Comment by DFurnes** [comment]
One open question – what do we do if Blink API requests fail? Do we catch the exception?

File: app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -14,10 +15,17 @@ class AppServiceProvider extends ServiceProvider
      */
     public function boot()
     {
-        User::creating(function ($user) {
+        User::creating(function (User $user) {
             // Set source automatically if not provided.
             $user->source = $user->source ?: client_id();
         });
+
+        User::created(function (User $user) {
```
Comment (deadlybutter) [inline app/Providers/AppServiceProvider.php]: is it worth keeping a log somewhere if this import fails?

Comment (weerd) [inline app/Providers/AppServiceProvider.php]: 🤔

Comment (DFurnes) [inline app/Providers/AppServiceProvider.php]: Yeah, I'm not sure! I'm tempted to say we queue this on our own end so it's non-blocking and something we can retry if Blink is down, but that seems to defeat the purpose of Blink.

Comment (DFurnes) [inline app/Providers/AppServiceProvider.php]: @sergii-tkachenko I'd love your thoughts here!

File: app/Models/User.php [hunk 2]
```diff
@@ -323,6 +323,34 @@ public static function drupalIDForNorthstarId($northstar_id)
...
+            'addr_city' => $this->addr_city,
+            'addr_state' => $this->addr_state,
+            'addr_zip' => $this->addr_zip,
+            'language' => $this->language,
+            'country' => $this->country,
+            'source' => $this->source,
+            'source_detail' => $this->source_detail,
+            'last_authenticated_at' => $this->last_authenticated_at ? $this->last_authenticated_at->toIso8601String() : null,
+            'updated_at' => $this->created_at->toIso8601String(),
+            'created_at' => $this->created_at->toIso8601String(),
```
Comment (sergii-tkachenko) [inline app/Models/User.php]: Could you also send `mobile_status`?

File: app/Providers/AppServiceProvider.php (hunk 1 above)
Comment (sergii-tkachenko) [inline app/Providers/AppServiceProvider.php]: Yes, you should just assume blink always returns a positive response. It's still possible that validation fails, or there are network connectivity problems, but that would be edge cases. 

In the first example, we'll be able to track malformed messages in blink logs.
Second, we should treat network connectivity issues as force majeure.

File: app/Models/User.php [hunk 3]
```diff
@@ -323,6 +323,34 @@ public static function drupalIDForNorthstarId($northstar_id)
...
+     */
+    public function toBlinkPayload()
+    {
+        return [
+            'id' => $this->id,
+            'email' => $this->email,
+            'mobile' => null, // <-- send once verified.
+            'first_name' => $this->first_name,
+            'last_name' => $this->last_name,
+            'addr_street1' => $this->addr_street1,
```
Comment (sergii-tkachenko) [inline app/Models/User.php]: We don't need `addr_street1` and `addr_street2`, you can exclude them.

File: app/Providers/AppServiceProvider.php (hunk 1 above)
Comment (DFurnes) [inline app/Providers/AppServiceProvider.php]: Okay, good to know!

File: app/Models/User.php (hunk 2 above)
Comment (DFurnes) [inline app/Models/User.php]: Done in 40f31dd!

File: app/Models/User.php (hunk 3 above)
Comment (DFurnes) [inline app/Models/User.php]: Done in 2237f38!

**Comment by DFurnes** [comment]
[(3) Northstar send a request to Blink to send a transactional message](https://trello.com/c/bpERY4m0/569-3-northstar-send-a-request-to-blink-to-send-a-transactional-message)

**Pull request merged**